.pre-commit-config.yaml

```
@@ -16,49 +18,53 @@ repos:
           - --allow-missing-credentials
       - id: detect-private-key
 
-  - repo: git://github.com/adrienverge/yamllint
-    rev: v1.25.0
+  - repo: https://github.com/adrienverge/yamllint
+    rev: v1.35.1
     hooks:
       - id: yamllint
         args:
           - -s
 
-  - repo: git://github.com/PyCQA/isort
-    rev: 5.6.4
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.13.2
     hooks:
       - id: isort
         args:
           - --check-only
           - --diff
 
-  - repo: git://github.com/PyCQA/prospector/
-    rev: 1.3.1
-    hooks:
-      - id: prospector
-        additional_dependencies:
-          - django
-          - ipaddress
-          - prospector
+  # This hook fails with the following error
+  # `pylint: failure / Tool pylint failed to run (exception was raised)`
+  # It's an old version which needs more work to be brought up to date
+  # - repo: https://github.com/PyCQA/prospector/
+  #   rev: 1.3.1
+  #   hooks:
+  #     - id: prospector
+  #       additional_dependencies:
+  #         - django
+  #         - ipaddress
+  #         - prospector
+  #         - setuptools
```

Updating prospector to the latest version works for me:

```suggestion
  - repo: https://github.com/prospector-dev/prospector/
    rev: v1.16.0
    hooks:
      - id: prospector
        additional_dependencies:
          - django
```

pylint complains about some code quality rules afterwards, but those should be fixable.

---

I was trying to avoid getting too many unrelated fixes in this PR, but I can add them if you don't mind. Or I can submit a PR to fix the CI before getting further on this? Let me know if you have a preference

---

I've extracted this in https://github.com/Dunedan/django-lockdown/pull/23 and hit a bit of blocker with prospector. I might try again later but I'm not familiar with the tool so any suggestions/pointers would be appreciated. Thanks

---

> I was trying to avoid getting too many unrelated fixes in this PR, but I can add them if you don't mind. Or I can submit a PR to fix the CI before getting further on this? Let me know if you have a preference

I'm fine with both approaches, but splitting different topics into different PRs is probably the easier way. :+1: